### ISSUE TITLE
SOLVED - SSLError(SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),),)

### ISSUE THREAD

**Issue by akuardit** [conda/conda#4930]
cannot install anything, create environment, or conda update conda

conda update conda -v
Fetching package metadata ...An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

overtaking stderr and stdout

stderr and stdout yielding back

Current conda install:

               platform : win-64
          conda version : 4.3.14
       conda is private : False
      conda-env version : 4.3.14
    conda-build version : not installed
         python version : 3.6.0.final.0
       requests version : 2.12.4
       root environment : D:\dev\Anaconda3  (writable)
    default environment : D:\dev\Anaconda3
       envs directories : D:\dev\Anaconda3\envs
                          C:\Users\user\AppData\Local\conda\conda\envs
                          C:\Users\user\DOCUME~1\MOBAXT~1\home\.conda\envs
          package cache : D:\dev\Anaconda3\pkgs
                          C:\Users\user\AppData\Local\conda\conda\pkgs
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/conda-canary/win-64
                          https://conda.anaconda.org/conda-canary/noarch
            config file : C:\Users\user\DOCUME~1\MOBAXT~1\home\.condarc
           offline mode : False
             user-agent : conda/4.3.14 requests/2.12.4 CPython/3.6.0 Windows/10 Windows/10.0.14393

`$ D:\dev\Anaconda3\Scripts\conda-script.py update conda -v`

    Traceback (most recent call last):
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\contrib\pyopenssl.py", line 417, in wrap_socket
        cnx.do_handshake()
      File "D:\dev\Anaconda3\lib\site-packages\OpenSSL\SSL.py", line 1426, in do_handshake
        self._raise_ssl_error(self._ssl, result)
      File "D:\dev\Anaconda3\lib\site-packages\OpenSSL\SSL.py", line 1174, in _raise_ssl_error
        _raise_current_error()
      File "D:\dev\Anaconda3\lib\site-packages\OpenSSL\_util.py", line 48, in exception_from_error_queue
        raise exception_type(errors)
    OpenSSL.SSL.Error: [('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')]

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\connectionpool.py", line 594, in urlopen
        chunked=chunked)
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\connectionpool.py", line 350, in _make_request
        self._validate_conn(conn)
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\connectionpool.py", line 835, in _validate_conn
        conn.connect()
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\connection.py", line 323, in connect
        ssl_context=context)
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\util\ssl_.py", line 324, in ssl_wrap_socket
        return context.wrap_socket(sock, server_hostname=server_hostname)
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\contrib\pyopenssl.py", line 424, in wrap_socket
        raise ssl.SSLError('bad handshake: %r' % e)
    ssl.SSLError: ("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",)

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "D:\dev\Anaconda3\lib\site-packages\requests\adapters.py", line 423, in send
        timeout=timeout
      File "D:\dev\Anaconda3\lib\site-packages\requests\packages\urllib3\connectionpool.py", line 624, in urlopen
        raise SSLError(e)
    requests.packages.urllib3.exceptions.SSLError: ("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",)

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 138, in fetch_repodata_remote_request
        timeout=timeout)
      File "D:\dev\Anaconda3\lib\site-packages\requests\sessions.py", line 501, in get
        return self.request('GET', url, **kwargs)
      File "D:\dev\Anaconda3\lib\site-packages\requests\sessions.py", line 488, in request
        resp = self.send(prep, **send_kwargs)
      File "D:\dev\Anaconda3\lib\site-packages\requests\sessions.py", line 609, in send
        r = adapter.send(request, **kwargs)
      File "D:\dev\Anaconda3\lib\site-packages\requests\adapters.py", line 497, in send
        raise SSLError(e, request=request)
    requests.exceptions.SSLError: ("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",)

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "D:\dev\Anaconda3\lib\site-packages\conda\exceptions.py", line 573, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "D:\dev\Anaconda3\lib\site-packages\conda\cli\main.py", line 134, in _main
        exit_code = args.func(args, p)
      File "D:\dev\Anaconda3\lib\site-packages\conda\cli\main_update.py", line 65, in execute
        install(args, parser, 'update')
      File "D:\dev\Anaconda3\lib\site-packages\conda\cli\install.py", line 222, in install
        unknown=index_args['unknown'], prefix=prefix)
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\index.py", line 92, in get_index
        index = fetch_index(channel_priority_map, use_cache=use_cache)
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\index.py", line 111, in fetch_index
        repodatas = collect_all_repodata(use_cache, tasks)
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 74, in collect_all_repodata
        repodatas = _collect_repodatas_serial(use_cache, tasks)
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 463, in _collect_repodatas_serial
        for url, schan, pri in tasks]
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 463, in <listcomp>
        for url, schan, pri in tasks]
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 109, in func
        res = f(*args, **kwargs)
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 442, in fetch_repodata
        mod_etag_headers.get('_mod'))
      File "D:\dev\Anaconda3\lib\site-packages\conda\core\repodata.py", line 305, in fetch_repodata_remote_request
        e.response)
    conda.exceptions.CondaHTTPError: HTTP None None for url <https://conda.anaconda.org/conda-forge/win-64/repodata.json>
    Elapsed: None

    An HTTP error occurred when trying to retrieve this URL.
    HTTP errors are often intermittent, and a simple retry will get you on your way.
    SSLError(SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),),)

**Comment by kilojoules** [comment]
+1 

    $ conda create -n my_numba_env
    Fetching package metadata ...vim 
    
    CondaHTTPError: HTTP None None for url <https://repo.continuum.io/pkgs/free/osx-64/repodata.json.bz2>
    Elapsed: None
    
    An HTTP error occurred when trying to retrieve this URL.
    HTTP errors are often intermittent, and a simple retry will get you on your way.
    SSLError(SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),),)
    
    
    
    ~/working $ vim ~/.bash_profile 
    ~/working $ conda install numba
    Fetching package metadata ...
    
    CondaHTTPError: HTTP None None for url <https://repo.continuum.io/pkgs/free/osx-64/repodata.json.bz2>
    Elapsed: None
    
    An HTTP error occurred when trying to retrieve this URL.
    HTTP errors are often intermittent, and a simple retry will get you on your way.
    SSLError(SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),),)

**Comment by akuardit** [comment]
@kilojoules, doesn't work on my laptop.

SOLVED...
Iam using windows 10 pro, never work with Mobaxterm 9/10, so move to Cmder.
- download cacert.pem from https://curl.haxx.se/ca/cacert.pem, save to c:\Cmder\cacert.pem
- open,  c:\Cmder\config\user-profile.sh 
  add this line, ... export REQUESTS_CA_BUNDLE=/c/Cmder/cacert.pem ...
- run, Cmder ->new console -> {bash::bash as Admin}

now can do anything with conda

**Comment by schattenjagerPL** [comment]
Hi just one hint for people struggling with this problem

> Solved this with conda config --set ssl_verify False

This did not resolve problem for me. But i just noticed  that when you enter this operation then anaconda creates user config file (.condarc) in his home directory (.e.g c:\Users\username\ ) and inside a file ssl_verify was set to True. Change it manually, save file and try again. I`m just downloading packages so it had to work.

**Comment by Pijju789** [comment]
"conda config --set ssl_verify False" working fine for "conda update conda" command. but throwing me error for "pip install eli5"

**Comment by WolfDSCoder** [comment]
I am trying to fix the SSL bad handshake error using @akuardit suggestion from above post:
I'm also using Windows 10, but am not sure how to complete the above steps.

**DONE --** download cacert.pem from https://curl.haxx.se/ca/cacert.pem, save to c:\Cmder\cacert.pem

**DONE --** open, c:\Cmder\config\user-profile.sh

**CHANGED --** add this line, ... export REQUESTS_CA_BUNDLE=/c/Cmder/cacert.pem ...
This was giving errors that "export" doesn't exist. I used the following, which doesn't error:
@set REQUESTS_CA_BUNDLE=/c/Cmder/cacert.pem;%PATH%
**_Is this correct?_**

**DON'T KNOW HOW TO DO THIS** -- run, Cmder ->new console -> {bash::bash as Admin}
**_Do I need to install 'bash'? Or what command is this for the Cmder console?_**

Thank you for any assistance. I can't use Conda or Anaconda so this is a very frustrating error.

**Comment by Code4SAFrankie** [comment]
conda config --set ssl_verify False solves the Anaconda issue, but it does not solve the pip issue

**Comment by bmond** [comment]
Any resolution found for this?

**Comment by WolfDSCoder** [comment]
You might try it as an Administrator and see if that works. 

When you say the Anaconda Prompt option did not work for you, but you can see the command worked, is your system working? Does REQUESTS_CA_BUNDLE have the correct path displayed?

**Comment by ibelieveai** [comment]
> SOLVED.
> 
> The problem was that the environment variable REQUESTS_CA_BUNDLE was not set in the Anaconda Prompt shell profile, or any other command shell profile such as Git or Cmdr. Below are the steps I used to correct this in the various shells. You only have to set this variable in the shell you are using.
> 
> You can verify REQUESTS_CA_BUNDLE is set in the shell by typing 'conda info -s' then look at the REQUESTS_CA_BUNDLE entry in the list, which in my case was not set.
> 
> **For Anaconda Prompt** -- You need to enter the 'set' command each time since it is not saved. Copy and paste the following at the prompt, but change the {username} to your own computer.
> `set REQUESTS_CA_BUNDLE=C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem `
> 
> **For CMDR** -- (Cmder is a console emulator downloaded from cmder.net)
> Open the following profile file in a text editor: C:\cmder\config\user-profile.sh
> Then insert the following line:
> `export REQUESTS_CA_BUNDLE='C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem'`
> Then run Cmdr and open a new console -> {bash::bash as Admin}
> You can now use the Cmdr shell to run conda.
> 
> **For Git Bash shell**
> Open the following profile file in a text editor: C:\users{username}.bashrc
> Then insert the following line:
> `export REQUESTS_CA_BUNDLE='C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem'`
> You can now use the Git Bash shell to run conda.
> 
> Notes:
> If you don't have the cacert.pem file in the certifi folder mentioned above, you can download it from
> 
> If anyone knows how to set the environment variable REQUESTS_CA_BUNDLE to be permanent in the Anaconda Prompt, please let me know. In the meantime, I created a batch.bat file to set it each time and run the Anaconda Prompt from there.
> 
> Now conda works. Hurray!

Worked! but how can we export the REQUESTS_CA_BUNDLE permanently without exporting everytime?

**Comment by TFT-Ben** [comment]
> 
> 
> SOLVED.
> 
> The problem was that the environment variable REQUESTS_CA_BUNDLE was not set in the Anaconda Prompt shell profile, or any other command shell profile such as Git or Cmdr. Below are the steps I used to correct this in the various shells. You only have to set this variable in the shell you are using.
> 
> You can verify REQUESTS_CA_BUNDLE is set in the shell by typing 'conda info -s' then look at the REQUESTS_CA_BUNDLE entry in the list, which in my case was not set.
> 
> **For Anaconda Prompt** -- You need to enter the 'set' command each time since it is not saved. Copy and paste the following at the prompt, but change the {username} to your own computer.
> `set REQUESTS_CA_BUNDLE=C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem `
> 
> **For CMDR** -- (Cmder is a console emulator downloaded from cmder.net)
> Open the following profile file in a text editor: C:\cmder\config\user-profile.sh
> Then insert the following line:
> `export REQUESTS_CA_BUNDLE='C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem'`
> Then run Cmdr and open a new console -> {bash::bash as Admin}
> You can now use the Cmdr shell to run conda.
> 
> **For Git Bash shell**
> Open the following profile file in a text editor: C:\users{username}.bashrc
> Then insert the following line:
> `export REQUESTS_CA_BUNDLE='C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem'`
> You can now use the Git Bash shell to run conda.
> 
> Notes:
> If you don't have the cacert.pem file in the certifi folder mentioned above, you can download it from
> 
> If anyone knows how to set the environment variable REQUESTS_CA_BUNDLE to be permanent in the Anaconda Prompt, please let me know. In the meantime, I created a batch.bat file to set it each time and run the Anaconda Prompt from there.
> 
> Now conda works. Hurray!

Using anaconda prompt here.  Has anyone got a fix for this (without having to enter it every time)
I literally can't use the GUI anaconda navigator right now because of this SSL error, stuck using the prompt...

**Comment by mingwandroid** [comment]
You can always set an env var permanently in your system. Either from the Windows GUI or from setx. However this fix seems weird to me, you shouldn't need to mess with REQUESTS_CA_BUNDLE and advice from May of last year is likely only relevant to people running Anaconda Distribution frrom around that time.

If someone can give me modern reproduction steps for this then I'd appreciate it.

**Comment by TFT-Ben** [comment]
> 
> 
> You can always set an env var permanently in your system. Either from the Windows GUI or from setx from cmd.exe. However this fix seems weird to me, you shouldn't need to mess with REQUESTS_CA_BUNDLE and advice from May of last year is likely only relevant to people running Anaconda Distribution from around that time.
> 
> If someone can give me modern reproduction steps for this then I'd appreciate it.

I'm on a brand new Anaconda 3 install.  The REQUESTS_CA_BUNDLE path seems to be defaulting to a certificate file in my root C: drive, as does the SSL_CERT_FILE.  

Opening a new anaconda prompt and doing anything like creating a new environment, or calling conda update --all, results in this:

> SSLError(MaxRetryError('HTTPSConnectionPool(host=\'repo.anaconda.com\', port=443): Max retries exceeded with url: /pkgs/main/win-64/current_repodata.json (Caused by SSLError(SSLError("bad handshake: Error([(\'SSL routines\', \'tls_process_server_certificate\', \'certificate verify failed\')])")))'))

Setting REQUESTS_CA_BUNDLE to C:\users\{username}\anaconda3\Lib\site-packages\certifi\cacert.pem fixes this

**Comment by mingwandroid** [comment]
> I'm on a brand new Anaconda 3 install. The REQUESTS_CA_BUNDLE path seems to be defaulting to a certificate file (.crt) in my root C: drive, as does the SSL_CERT_FILE.

In what sense? We do not set this env. variable. Did you have it set in your system? Can you show the output of `conda list --show-channel-urls` when your installation is in this problematic state please? I have been looking at OpenSSL issues but they are not issues that are present in the Anaconda 2019.07 release, they are issues I introduced more recently than that (they're fixed now though).

**Comment by TFT-Ben** [comment]
Output of conda list --show-channel-urls:

```
# packages in environment at C:\Users\darkb\Anaconda3:
#
# Name                    Version                   Build  Channel
_anaconda_depends         2019.03                  py37_0    defaults
_ipyw_jlab_nb_ext_conf    0.1.0                    py37_0    defaults
alabaster                 0.7.12                   py37_0    defaults
anaconda                  custom                   py37_1    defaults
anaconda-client           1.7.2                    py37_0    defaults
anaconda-navigator        1.9.7                    py37_0    defaults
anaconda-project          0.8.3                      py_0    defaults
asn1crypto                0.24.0                   py37_0    defaults
astroid                   2.2.5                    py37_0    defaults
astropy                   3.2.1            py37he774522_0    defaults
atomicwrites              1.3.0                    py37_1    defaults
attrs                     19.1.0                   py37_1    defaults
babel                     2.7.0                      py_0    defaults
backcall                  0.1.0                    py37_0    defaults
backports                 1.0                        py_2    defaults
backports.functools_lru_cache 1.5                        py_2    defaults
backports.os              0.1.1                    py37_0    defaults
backports.shutil_get_terminal_size 1.0.0                    py37_2    defaults
backports.tempfile        1.0                        py_1    defaults
backports.weakref         1.0.post1                  py_1    defaults
beautifulsoup4            4.8.0                    py37_0    defaults
bitarray                  1.0.1            py37he774522_0    defaults
bkcharts                  0.2                      py37_0    defaults
blas                      1.0                         mkl    defaults
bleach                    3.1.0                    py37_0    defaults
blosc                     1.16.3               h7bd577a_0    defaults
bokeh                     1.3.4                    py37_0    defaults
boto                      2.49.0                   py37_0    defaults
bottleneck                1.2.1            py37h452e1ab_1    defaults
bzip2                     1.0.8                he774522_0    defaults
ca-certificates           2019.5.15                     1    defaults
certifi                   2019.6.16                py37_1    defaults
cffi                      1.12.3           py37h7a1dbc1_0    defaults
chardet                   3.0.4                 py37_1003    defaults
click                     7.0                      py37_0    defaults
cloudpickle               1.2.2                      py_0    defaults
clyent                    1.2.2                    py37_1    defaults
colorama                  0.4.1                    py37_0    defaults
comtypes                  1.1.7                    py37_0    defaults
conda                     4.7.11                   py37_0    defaults
conda-build               3.18.9                   py37_3    defaults
conda-env                 2.6.0                         1    defaults
conda-package-handling    1.3.11                   py37_0    defaults
conda-verify              3.4.2                      py_1    defaults
console_shortcut          0.1.1                         3    defaults
contextlib2               0.5.5                    py37_0    defaults
cryptography              2.7              py37h7a1dbc1_0    defaults
curl                      7.65.3               h2a8f88b_0    defaults
cycler                    0.10.0                   py37_0    defaults
cython                    0.29.13          py37ha925a31_0    defaults
cytoolz                   0.10.0           py37he774522_0    defaults
dask                      2.3.0                      py_0    defaults
dask-core                 2.3.0                      py_0    defaults
decorator                 4.4.0                    py37_1    defaults
defusedxml                0.6.0                      py_0    defaults
distributed               2.3.2                      py_1    defaults
docutils                  0.15.2                   py37_0    defaults
entrypoints               0.3                      py37_0    defaults
et_xmlfile                1.0.1                    py37_0    defaults
fastcache                 1.1.0            py37he774522_0    defaults
filelock                  3.0.12                     py_0    defaults
flask                     1.1.1                      py_0    defaults
freetype                  2.9.1                ha9979f8_1    defaults
fsspec                    0.4.0                      py_0    defaults
future                    0.17.1                   py37_0    defaults
get_terminal_size         1.0.0                h38e98db_0    defaults
gevent                    1.4.0            py37he774522_0    defaults
glob2                     0.7                        py_0    defaults
greenlet                  0.4.15           py37hfa6e2cd_0    defaults
h5py                      2.9.0            py37h5e291fa_0    defaults
hdf5                      1.10.4               h7ebc959_0    defaults
heapdict                  1.0.0                    py37_2    defaults
html5lib                  1.0.1                    py37_0    defaults
icc_rt                    2019.0.0             h0cc432a_1    defaults
icu                       58.2                 ha66f8fd_1    defaults
idna                      2.8                      py37_0    defaults
imageio                   2.5.0                    py37_0    defaults
imagesize                 1.1.0                    py37_0    defaults
importlib_metadata        0.19                     py37_0    defaults
intel-openmp              2019.4                      245    defaults
ipykernel                 5.1.2            py37h39e3cac_0    defaults
ipython                   7.8.0            py37h39e3cac_0    defaults
ipython_genutils          0.2.0                    py37_0    defaults
ipywidgets                7.5.1                      py_0    defaults
isort                     4.3.21                   py37_0    defaults
itsdangerous              1.1.0                    py37_0    defaults
jdcal                     1.4.1                      py_0    defaults
jedi                      0.15.1                   py37_0    defaults
jinja2                    2.10.1                   py37_0    defaults
joblib                    0.13.2                   py37_0    defaults
jpeg                      9b                   hb83a4c4_2    defaults
json5                     0.8.5                      py_0    defaults
jsonschema                3.0.2                    py37_0    defaults
jupyter                   1.0.0                    py37_7    defaults
jupyter_client            5.3.1                      py_0    defaults
jupyter_console           6.0.0                    py37_0    defaults
jupyter_core              4.5.0                      py_0    defaults
jupyterlab                1.1.3              pyhf63ae98_0    defaults
jupyterlab_server         1.0.0                      py_1    defaults
keyring                   18.0.0                   py37_0    defaults
kiwisolver                1.1.0            py37ha925a31_0    defaults
krb5                      1.16.1               hc04afaa_7    defaults
lazy-object-proxy         1.4.2            py37he774522_0    defaults
libarchive                3.3.3                h0643e63_5    defaults
libcurl                   7.65.3               h2a8f88b_0    defaults
libiconv                  1.15                 h1df5818_7    defaults
liblief                   0.9.0                ha925a31_2    defaults
libpng                    1.6.37               h2a8f88b_0    defaults
libsodium                 1.0.16               h9d3ae62_0    defaults
libssh2                   1.8.2                h7a1dbc1_0    defaults
libtiff                   4.0.10               hb898794_2    defaults
libxml2                   2.9.9                h464c3ec_0    defaults
libxslt                   1.1.33               h579f668_0    defaults
llvmlite                  0.29.0           py37ha925a31_0    defaults
locket                    0.2.0                    py37_1    defaults
lxml                      4.4.1            py37h1350720_0    defaults
lz4-c                     1.8.1.2              h2fa13f4_0    defaults
lzo                       2.10                 h6df0209_2    defaults
m2w64-gcc-libgfortran     5.3.0                         6    defaults
m2w64-gcc-libs            5.3.0                         7    defaults
m2w64-gcc-libs-core       5.3.0                         7    defaults
m2w64-gmp                 6.1.0                         2    defaults
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    defaults
markupsafe                1.1.1            py37he774522_0    defaults
matplotlib                3.1.1            py37hc8f65d3_0    defaults
mccabe                    0.6.1                    py37_1    defaults
menuinst                  1.4.16           py37he774522_0    defaults
mistune                   0.8.4            py37he774522_0    defaults
mkl                       2019.4                      245    defaults
mkl-service               2.3.0            py37hb782905_0    defaults
mkl_fft                   1.0.14           py37h14836fe_0    defaults
mkl_random                1.0.2            py37h343c172_0    defaults
mock                      3.0.5                    py37_0    defaults
more-itertools            7.2.0                    py37_0    defaults
mpmath                    1.1.0                    py37_0    defaults
msgpack-python            0.6.1            py37h74a9793_1    defaults
msys2-conda-epoch         20160418                      1    defaults
multipledispatch          0.6.0                    py37_0    defaults
navigator-updater         0.2.1                    py37_0    defaults
nbconvert                 5.5.0                      py_0    defaults
nbformat                  4.4.0                    py37_0    defaults
networkx                  2.3                        py_0    defaults
nltk                      3.4.5                    py37_0    defaults
nose                      1.3.7                    py37_2    defaults
notebook                  6.0.1                    py37_0    defaults
numba                     0.45.1           py37hf9181ef_0    defaults
numexpr                   2.7.0            py37hdce8814_0    defaults
numpy                     1.16.5           py37h19fb1c0_0    defaults
numpy-base                1.16.5           py37hc3f5095_0    defaults
numpydoc                  0.9.1                      py_0    defaults
olefile                   0.46                     py37_0    defaults
openpyxl                  2.6.2                      py_0    defaults
openssl                   1.1.1d               he774522_0    defaults
packaging                 19.1                     py37_0    defaults
pandas                    0.25.1           py37ha925a31_0    defaults
pandoc                    2.2.3.2                       0    defaults
pandocfilters             1.4.2                    py37_1    defaults
parso                     0.5.1                      py_0    defaults
partd                     1.0.0                      py_0    defaults
path.py                   12.0.1                     py_0    defaults
pathlib2                  2.3.4                    py37_0    defaults
patsy                     0.5.1                    py37_0    defaults
pep8                      1.7.1                    py37_0    defaults
pickleshare               0.7.5                    py37_0    defaults
pillow                    6.1.0            py37hdc69c19_0    defaults
pip                       19.2.2                   py37_0    defaults
pkginfo                   1.5.0.1                  py37_0    defaults
pluggy                    0.12.0                     py_0    defaults
ply                       3.11                     py37_0    defaults
powershell_shortcut       0.0.1                         2    defaults
prometheus_client         0.7.1                      py_0    defaults
prompt_toolkit            2.0.9                    py37_0    defaults
psutil                    5.6.3            py37he774522_0    defaults
py                        1.8.0                    py37_0    defaults
py-lief                   0.9.0            py37ha925a31_2    defaults
pycodestyle               2.5.0                    py37_0    defaults
pycosat                   0.6.3            py37hfa6e2cd_0    defaults
pycparser                 2.19                     py37_0    defaults
pycrypto                  2.6.1            py37hfa6e2cd_9    defaults
pycurl                    7.43.0.3         py37h7a1dbc1_0    defaults
pyflakes                  2.1.1                    py37_0    defaults
pygments                  2.4.2                      py_0    defaults
pylint                    2.3.1                    py37_0    defaults
pyodbc                    4.0.27           py37ha925a31_0    defaults
pyopenssl                 19.0.0                   py37_0    defaults
pyparsing                 2.4.2                      py_0    defaults
pyqt                      5.9.2            py37h6538335_2    defaults
pyreadline                2.1                      py37_1    defaults
pyrsistent                0.14.11          py37he774522_0    defaults
pysocks                   1.7.0                    py37_0    defaults
pytables                  3.5.2            py37h1da0976_1    defaults
pytest                    5.0.1                    py37_0    defaults
pytest-arraydiff          0.3              py37h39e3cac_0    defaults
pytest-astropy            0.5.0                    py37_0    defaults
pytest-doctestplus        0.3.0                    py37_0    defaults
pytest-openfiles          0.3.2                    py37_0    defaults
pytest-remotedata         0.3.2                    py37_0    defaults
python                    3.7.4                h5263a28_0    defaults
python-dateutil           2.8.0                    py37_0    defaults
python-libarchive-c       2.8                     py37_13    defaults
pytz                      2019.2                     py_0    defaults
pywavelets                1.0.3            py37h8c2d366_1    defaults
pywin32                   223              py37hfa6e2cd_1    defaults
pywinpty                  0.5.5                 py37_1000    defaults
pyyaml                    5.1.2            py37he774522_0    defaults
pyzmq                     18.1.0           py37ha925a31_0    defaults
qt                        5.9.7            vc14h73c81de_0    defaults
qtawesome                 0.5.7                    py37_1    defaults
qtconsole                 4.5.5                      py_0    defaults
qtpy                      1.9.0                      py_0    defaults
requests                  2.22.0                   py37_0    defaults
rope                      0.14.0                     py_0    defaults
ruamel_yaml               0.15.46          py37hfa6e2cd_0    defaults
scikit-image              0.15.0           py37ha925a31_0    defaults
scikit-learn              0.21.2           py37h6288b17_0    defaults
scipy                     1.3.1            py37h29ff71c_0    defaults
seaborn                   0.9.0                    py37_0    defaults
send2trash                1.5.0                    py37_0    defaults
setuptools                41.0.1                   py37_0    defaults
simplegeneric             0.8.1                    py37_2    defaults
singledispatch            3.4.0.3                  py37_0    defaults
sip                       4.19.8           py37h6538335_0    defaults
six                       1.12.0                   py37_0    defaults
snappy                    1.1.7                h777316e_3    defaults
snowballstemmer           1.9.0                      py_0    defaults
sortedcollections         1.1.2                    py37_0    defaults
sortedcontainers          2.1.0                    py37_0    defaults
soupsieve                 1.9.2                    py37_0    defaults
sphinx                    2.1.2                      py_0    defaults
sphinxcontrib             1.0                      py37_1    defaults
sphinxcontrib-applehelp   1.0.1                      py_0    defaults
sphinxcontrib-devhelp     1.0.1                      py_0    defaults
sphinxcontrib-htmlhelp    1.0.2                      py_0    defaults
sphinxcontrib-jsmath      1.0.1                      py_0    defaults
sphinxcontrib-qthelp      1.0.2                      py_0    defaults
sphinxcontrib-serializinghtml 1.1.3                      py_0    defaults
sphinxcontrib-websupport  1.1.2                      py_0    defaults
spyder                    3.3.6                    py37_0    defaults
spyder-kernels            0.5.1                    py37_0    defaults
sqlalchemy                1.3.7            py37he774522_0    defaults
sqlite                    3.29.0               he774522_0    defaults
statsmodels               0.10.1           py37h8c2d366_0    defaults
sympy                     1.4                      py37_0    defaults
tbb                       2019.4               h74a9793_0    defaults
tblib                     1.4.0                      py_0    defaults
terminado                 0.8.2                    py37_0    defaults
testpath                  0.4.2                    py37_0    defaults
tk                        8.6.8                hfa6e2cd_0    defaults
toolz                     0.10.0                     py_0    defaults
tornado                   6.0.3            py37he774522_0    defaults
tqdm                      4.32.1                     py_0    defaults
traitlets                 4.3.2                    py37_0    defaults
unicodecsv                0.14.1                   py37_0    defaults
urllib3                   1.24.2                   py37_0    defaults
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.16.27012          hf0eaf9b_0    defaults
wcwidth                   0.1.7                    py37_0    defaults
webencodings              0.5.1                    py37_1    defaults
werkzeug                  0.15.5                     py_0    defaults
wheel                     0.33.4                   py37_0    defaults
widgetsnbextension        3.5.1                    py37_0    defaults
win_inet_pton             1.1.0                    py37_0    defaults
win_unicode_console       0.5                      py37_0    defaults
wincertstore              0.2                      py37_0    defaults
winpty                    0.4.3                         4    defaults
wrapt                     1.11.2           py37he774522_0    defaults
xlrd                      1.2.0                    py37_0    defaults
xlsxwriter                1.1.8                      py_0    defaults
xlwings                   0.15.8                   py37_0    defaults
xlwt                      1.3.0                    py37_0    defaults
xz                        5.2.4                h2fa13f4_4    defaults
yaml                      0.1.7                hc54c509_2    defaults
zeromq                    4.3.1                h33f27b4_3    defaults
zict                      1.0.0                      py_0    defaults
zipp                      0.5.2                      py_0    defaults
zlib                      1.2.11               h62dcd97_3    defaults
zstd                      1.3.7                h508b16e_0    defaults
```

Let me check my system env variables

**Comment by TFT-Ben** [comment]
Ah... I have REQUESTS_CA_BUNDLE set at system level... 
I don't remember why I would have that set - is that something I can remove and expect conda to function or does it need pointing at cacert.pem?

**Comment by mingwandroid** [comment]
I suspected as much.

Well, the Anaconda Distribution is a general purpose distribution. We provide lots of software. We are not at liberty to modify the bahaviour of said software (e.g. remoing support for the REQUESTS_CA_BUNDLE env var from, I guess, the requests package).

If we start doing that then people and software with legitimate reasons to set that variable will complain and fail respectively.

All we can really do IMHO is to catch these conditions after they've happened and craft exception messages that are very helpful as to the likely cause.

**Comment by mingwandroid** [comment]
> is that something I can remove and expect conda to function or does it need pointing at cacert.pem?

If you unset this then yes, you can expect Anaconda software to do the right thing. This env. var is an override, preventing sensible defaults from being used.

Unsetting it will possibly cause some other (less well engineered; from this perspective!) software you have installed to not work though, so you may want to make a batch script that calls conda activate but also unsets this variable. It's up to you though really.

**Comment by TFT-Ben** [comment]
Thanks! I’ve set something up that appears to be working. ---- On Sat, 14 Sep 2019 18:48:58 +0100  notifications@github.com<notifications@github.com> wrote ---- 
is that something I can remove and expect conda to function or does it need pointing at cacert.pem? 
 
If you unset this then yes, you can expect Anaconda software to do the right thing. This env. var is an override, preventing sensible defaults from being used. 
Unsetting it will possibly cause some other (less well engineered; from this perspective!) software you have installed to not work though, so you may want to make a batch script that calls conda activate but also unsets this variable. It's up to you though really.

**Comment by adithyaonline** [comment]
For folks behind a corporate/university network with proxies etc who are still facing issues with this,

I managed to get it working, 
steps detailed [here](https://github.com/ContinuumIO/anaconda-issues/issues/3292#issuecomment-569237742)

**Comment by Oroelipas** [comment]
SOLUTION (it worked for me):
Check the time and date of your computer is correct. SSL can give problems if not

**Comment by iver56** [comment]
Thanks @Oroelipas, that was indeed the problem. I restarted my Docker host, and the problem disappeared.

**Comment by rg2609** [comment]
`authlib.integrations.base_client.errors.OAuthError: unauthorized_client: Invalid client secret`
I am getting this error
